src/ape/api/providers.py

```diff
@@ -271,8 +325,39 @@ def priority_fee(self) -> int:
 
     @property
     def base_fee(self) -> int:
-        block = self._web3.eth.get_block("latest")
-        return block.baseFeePerGas  # type: ignore
+        """
+        Returns the current base fee from the latest block.
+
+        NOTE: If your chain does not support base_fees (EIP-1559),
+        this method returns 0.
+        """
+        block = self.get_block("latest")
+        return block.gas_data.base_fee or 0
+
+    def get_block(
+        self, block_id: Union[str, int, HexBytes, Literal["latest"], Literal["pending"]]
```

Comment:
was thinking to add this union to `ape.types` as `BlockId = Union[str, int, HexBytes, Literal["latest"], Literal["pending"]`